src/mam/mod_mam_utils.erl

```diff
@@ -1155,19 +1156,75 @@ is_policy_violation(TotalCount, Offset, MaxResultLimit, LimitPassed) ->
       LookupResult :: mod_mam:lookup_result(),
       R :: {ok, mod_mam:lookup_result()} | {error, item_not_found}.
 check_for_item_not_found(#rsm_in{direction = before, id = ID},
-                         PageSize, {TotalCount, Offset, MessageRows}) ->
+                         _PageSize, {TotalCount, Offset, MessageRows}) ->
```

Comment:
If this parameter is not needed it can be removed, maybe?

Reply:
of course xd
Remained for compat.
Will get refactored with the tuple. And we probably wanna pass Params, instead of just RSM.